### ISSUE TITLE
While analysing an ongoing game, the end of the game stops the analysis too

### ISSUE THREAD

**Issue by Stooh** [online-go/online-go.com#85]
### Summary

While watching an ongoing game, and navigating it / analysing it, the process is often disrupted by what is currently happening in the game. The worst case being the end of the game which always force the user to the end of game screen, ending any ongoing analysis.

### Environment

 * Operating system: Linux
 * Browser: Firefox
 * Browser Version: 51.0.1 (64 bits)

### Steps To Reproduce:

1. Go to the "Watch" tab
2. Click on a game that is close to finishing to watch it
3. Navigate back a few moves (with left arrow)
4. You are now in analysis mode
5. Wait for the end of the game
6. See that you are now out of analysis mode, and instead on the end game screen, while you should still be in analysis mode

**Comment by anoek** [comment]
Note that any changes to the current behavior needs to ensure that users are made aware of any new moves or change in state of a game. This is more important for players than spectators. We also may want to consider keeping the current behavior for players, while allowing spectators to remain in analysis mode as moves are played out.

**Comment by Stooh** [comment]
You're right : a player *should* be forced out of analysis mode when the end of the game occurs (also when counting points), which is the current behaviour. I didn't consider this case. 

However, I remember experiencing weird bugs while trying to stay in analysis mode during the end of games. For example, I *think* that entering analysis mode during the scoring phase ends up displaying both the analysis mode and the scoring phase (i.e. black/white territories of the end of the game are drawn on top of analysis mode). If the current behaviour is kept for players, there might be some bugs lying around.

The current behaviour for spectators (and maybe players ?), is that you hear every move played even though you don't see them (also the game timeline is updated with the new moves) : which seems about right. Only events related to the "end of the game" seem to force a spectator out of analysis mode.

**Comment by pdg137** [comment]
Related to this, the behavior of the forward/back buttons is extremely weird in the scoring phase:

![Screenshot_20211207-073123](https://user-images.githubusercontent.com/466760/145059225-d4519708-0106-4397-82f8-8c7d5aac4885.png)

In fact forward/back and analysis would be particularly useful during scoring, when players might want to start reviewing or analyze status of the final groups. But the display is so strange that it's not really usable.

**Comment by anoek** [comment]
Suggestions on improvements are often more useful than simply saying it's strange and not usable

**Comment by benjaminpjones** [comment]
+1 to suggestions on improvements, but I may be able to speak to the strangeness/unusability from the previous comment.  During the stone removal phase, if I click the analyze button (or the fast-forward/play buttons), I am able to move backward and forward, but I am not able to lay out variations as I would expect in analyze mode.  Clicking an intersection simply toggles the score markers for the surrounding area rather than places a variation stone.  IMO, clicking analysis should enter full on analysis mode, without the score shown (sort of how it works after the game has finished).

Screen recording: https://user-images.githubusercontent.com/25233703/145105151-8062d17e-03ac-46d2-911b-7726850db3c1.mov

This is particularly noticeable when there is a disagreement on dead stones.  I'd like to share a "kill variation", but can't do that because there is no way for me to play out a variation.

**Comment by anoek** [comment]
Oh yeah that's horrible

**Comment by benjaminpjones** [comment]
Linking #1109, the issue @pdg137 referenced

**Comment by pdg137** [comment]
As far as I know it's still a major issue
